pages/admin/directory.tsx

```diff
@@ -516,17 +479,87 @@ const MemberEdit: FC<{
     }
   };
 
-  const handleRegionChange = (value: string) => {
-    if (value === "New") {
-      console.log("create new region");
-      return;
+  const updateMemberField = async (uid: string, fieldName: string, newData: any, suggestedFilter?: boolean) => {
+    const response = await fetch("/api/update-member", {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${await user.getIdToken()}`,
+      },
+      body: JSON.stringify({
+        uid: uid,
+        fieldName: fieldName,
+        newData: newData,
+        currentUser: user.displayName || user.uid,
+        suggestedFilter: suggestedFilter,
+      }),
+    });
+    if (response.status !== 200) {
+      throw new Error(`Error updating ${fieldName} for ${uid}`);
+    }
+    console.log(`✅ updated ${fieldName} in firebase for ${uid}`);
+    return response;
+  };
+
+  const updateSecureEmail = async (uid: string, email: string) => {
+    const response = await fetch("/api/update-email-by-id", {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${await user.getIdToken()}`,
+      },
+      body: JSON.stringify({
+        uid: uid,
+        email: email,
+        currentUser: user.displayName || user.uid,
+      }),
+    });
+    if (response.status !== 200) {
+      throw new Error(`Error updating email for ${uid}`);
+    }
+    console.log(`✅ updated email in firebase for ${uid}`);
+    return response;
+  };
+
+  const saveChanges = async () => {
+    name !== member.name && (await updateMemberField(member.id, mFields.NAME, name));
```

Comment:
This block is pretty large, wondering if we could use `lodash.isEqual` and if not, then send a single PUT request to the API.

Reply:
agreed, I think this can be simplified, will be adding this in the refactor for PUT

Reply:
done with the single PUT request for member fields